modules/recetas/recetasController.ts

```diff
@@ -1,15 +1,68 @@
+import * as moment from 'moment';
 import { Types } from 'mongoose';
 import { Auth } from '../../auth/auth.class';
 import { searchMatriculas } from '../../core/tm/controller/profesional';
 import { MotivosReceta, Receta } from './receta-schema';
-import { ParamsIncorrect, RecetaNotFound, RecetaNotEdit } from './recetas.error';
-import * as moment from 'moment';
+import { createLog, informarLog, updateLog } from './recetaLogs';
+import { ParamsIncorrect, RecetaNotEdit, RecetaNotFound } from './recetas.error';
 import { getReceta } from './services/receta';
-import { updateLog, informarLog, createLog } from './recetaLogs';
+
+export async function consultarDispensa(receta, sistema) {
+    try {
+        // Consultar estado en el sistema externo
+        const pacienteId = receta.paciente.id;
+        const recetaDisp = await getReceta(Types.ObjectId(receta.id), pacienteId, sistema);
```

Comment:
Por algun motivo este get falla bastante pasandole el id del paciente como bson. Dejo sugerencia que corrigió todas las ocurrencias:
```suggestion
        const pacienteId = receta.paciente.id.toString();
        const recetaDisp = await getReceta(Types.ObjectId(receta.id), Types.ObjectId(pacienteId), sistema);
```